### PR TITLE
CBL-4133: Rewrite Xamarin Android reachability class to use Xamarin Essentials

### PR DESCRIPTION
**Pull request by borrrden**
To work around a bug with NetworkAddressChanged in which socket fds are not correctly freed by mono

Companion PR: https://github.com/couchbaselabs/couchbase-lite-net-ee/pull/97

### CODE REVIEW

File: src/Couchbase.Lite.Shared/Support/Android/XEReachability.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        {
+            Console.WriteLine($"XEReachability changed: {e.NetworkAccess}");
+            var state = NetworkReachabilityStatus.Unknown;
+            if (e.NetworkAccess == NetworkAccess.None) {
+                state = NetworkReachabilityStatus.Unreachable;
+            } else if(e.NetworkAccess != NetworkAccess.Unknown) {
+                state = NetworkReachabilityStatus.Reachable;
+            }
+
+            var eventArgs = new NetworkReachabilityChangeEventArgs(state);
```
Comment (bmeike) [inline src/Couchbase.Lite.Shared/Support/Android/XEReachability.cs:45]: Stylistic question, here.  Why the variable?  Could this be written:
StatusChanged?.Invoke(this, new NetworkReachabilityChangeEventArgs(state));

File: src/Couchbase.Lite.Shared/Support/Android/XEReachability.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+using Xamarin.Essentials;
+
+namespace Couchbase.Lite.Support
+{
+    [CouchbaseDependency]
+    internal sealed class XEReachability : IReachability
+    {
+        public Uri Url { get; set; }
+
+        public event EventHandler<NetworkReachabilityChangeEventArgs> StatusChanged;
```
Comment (bmeike) [inline src/Couchbase.Lite.Shared/Support/Android/XEReachability.cs:31]: What is an "event"?

**Review by bmeike — APPROVED** [review]
Looks good as far as I can see....

File: src/Couchbase.Lite.Shared/Support/Android/XEReachability.cs (hunk 1 above)
Comment (borrrden) [reply src/Couchbase.Lite.Shared/Support/Android/XEReachability.cs:45]: For me it is just more readable this way

**Pull request merged**